src/java/htsjdk/samtools/PrintReads.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2010 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ */
+package htsjdk.samtools;
+
+
+import htsjdk.samtools.util.Log;
+import htsjdk.samtools.util.ProgressLogger;
+
+import java.io.File;
+import java.io.IOException;
+
+/**
+ * This is a example program showing how to use SAM readers and writers.
+ * It's also useful for measuring time.
+ */
+public final class PrintReads {
+    private PrintReads(){}
+
+    private static final Log log = Log.getInstance(PrintReads.class);
+
+    public static void main(String[] args) throws IOException {
+        if (args.length != 3){
+            System.out.println("Usage: PrintReads inFile decodeEverything outFile");
+            System.exit(1);
+        }
+        final File inputFile = new File(args[0]);
+        final boolean decodeEverything = Boolean.parseBoolean(args[1]); //useful to test (realistic) scenarios in which every record is always fully decoded.
+        final File outputFile = new File(args[2]);
+
+        try( SamReader reader = SamReaderFactory.makeDefault().validationStringency(ValidationStringency.SILENT).open(inputFile)) {
+            final SAMFileHeader header = reader.getFileHeader();
+            try (SAMFileWriter writer = new SAMFileWriterFactory().makeBAMWriter(header, true, outputFile)) {
+                final ProgressLogger pl= new ProgressLogger(log, 100000);
```

Comment:
Does the progress logger emit time upon completion? if not, might as well do that from the program rather than have to wrap this with an external "time" command.